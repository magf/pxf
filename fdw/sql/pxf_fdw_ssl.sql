-- ===================================================================
-- Validation for SSL options
-- ===================================================================
-- This is expected to be run in the container
-- 
-- start_matchignore
-- m/^WARNING:  skipping ".*" --- cannot analyze this foreign table/
-- end_matchignore

CREATE FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl
    HANDLER pxf_fdw_handler
    VALIDATOR pxf_fdw_validator
    OPTIONS (protocol 'system', mpp_execute 'all segments',
        pxf_protocol 'https',
        pxf_ssl_cacert '/opt/ssl/certs/ca-cert',
        pxf_ssl_cert   '/opt/ssl/certs/pxf-client.pem',
        pxf_ssl_cert_type 'PEM',
        pxf_ssl_key '/opt/ssl/certs/pxf-client.key'
    );

CREATE SERVER pxf_filter_push_down_server_ssl
    FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl;

CREATE USER MAPPING FOR CURRENT_USER SERVER pxf_filter_push_down_server_ssl;

CREATE FOREIGN TABLE test_filter(
    t0 text,
    a1 integer,
    b2 boolean,
    c3 numeric,
    d4 char(2),
    e5 varchar(2),
    x1 bpchar(2),
    x2 smallint,
    x3 bigint,
    x4 real,
    x5 float8,
    x6 bytea,
    x7 date,
    x8 time,
    x9 timestamp,
    x10 timestamp with time zone,
    x11 interval,
    x12 uuid,
    x13 json,
    x14 jsonb,
    x15 int2[],
    x16 int4[],
    x17 int8[],
    x18 bool[],
    x19 text[],
    x20 float4[],
    x21 float8[],
    x22 bytea[],
    x23 bpchar[],
    x24 varchar(2)[],
    x25 date[],
    x26 uuid[],
    x27 numeric[],
    x28 time[],
    x29 timestamp[],
    x30 timestamp with time zone[],
    x31 interval[],
    x32 json[],
    x33 jsonb[],
    filterValue text)
    SERVER pxf_filter_push_down_server_ssl
    OPTIONS (resource 'dummy_path', format 'filter', delimiter ',');

SELECT t0, a1 FROM test_filter;

ALTER FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl
    OPTIONS(ADD pxf_ssl_verify_peer 'false');

-- Check writable table

CREATE FOREIGN DATA WRAPPER file_pxf_fdw_ssl
    HANDLER pxf_fdw_handler
    VALIDATOR pxf_fdw_validator
    OPTIONS(protocol 'file', mpp_execute 'all segments',
        pxf_protocol 'https',
        pxf_ssl_cacert '/opt/ssl/certs/ca-cert',
        pxf_ssl_cert   '/opt/ssl/certs/pxf-client.pem',
        pxf_ssl_cert_type 'PEM',
        pxf_ssl_key '/opt/ssl/certs/pxf-client.key'
    );

CREATE SERVER pxf_file_server_ssl
    FOREIGN DATA WRAPPER file_pxf_fdw_ssl OPTIONS(
    config 'default');

CREATE USER MAPPING FOR CURRENT_USER SERVER pxf_file_server_ssl;

CREATE FOREIGN TABLE test_file2 (
    id int,
    descr text
) SERVER pxf_file_server_ssl
OPTIONS (resource 'fdw_file', format 'csv', delimiter ',');

INSERT INTO test_file2 select id, md5('hello'||id::text)as descr from generate_series(1,10) as id;

\! find /tmp/pxf/fdw_file -type f -name \[0-9]\* | xargs cat | LC_ALL=C sort -t',' -k1,1n

