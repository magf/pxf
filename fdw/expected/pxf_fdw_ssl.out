-- ===================================================================
-- Validation for SSL options
-- ===================================================================
-- This is expected to be run in the container
-- 
-- start_matchignore
-- m/^WARNING:  skipping ".*" --- cannot analyze this foreign table/
-- end_matchignore
CREATE FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl
    HANDLER pxf_fdw_handler
    VALIDATOR pxf_fdw_validator
    OPTIONS (protocol 'system', mpp_execute 'all segments',
        pxf_protocol 'https',
        pxf_ssl_cacert '/opt/ssl/certs/ca-cert',
        pxf_ssl_cert   '/opt/ssl/certs/pxf-client.pem',
        pxf_ssl_cert_type 'PEM',
        pxf_ssl_key '/opt/ssl/certs/pxf-client.key'
    );
CREATE SERVER pxf_filter_push_down_server_ssl
    FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl;
CREATE USER MAPPING FOR CURRENT_USER SERVER pxf_filter_push_down_server_ssl;
CREATE FOREIGN TABLE test_filter(
    t0 text,
    a1 integer,
    b2 boolean,
    c3 numeric,
    d4 char(2),
    e5 varchar(2),
    x1 bpchar(2),
    x2 smallint,
    x3 bigint,
    x4 real,
    x5 float8,
    x6 bytea,
    x7 date,
    x8 time,
    x9 timestamp,
    x10 timestamp with time zone,
    x11 interval,
    x12 uuid,
    x13 json,
    x14 jsonb,
    x15 int2[],
    x16 int4[],
    x17 int8[],
    x18 bool[],
    x19 text[],
    x20 float4[],
    x21 float8[],
    x22 bytea[],
    x23 bpchar[],
    x24 varchar(2)[],
    x25 date[],
    x26 uuid[],
    x27 numeric[],
    x28 time[],
    x29 timestamp[],
    x30 timestamp with time zone[],
    x31 interval[],
    x32 json[],
    x33 jsonb[],
    filterValue text)
    SERVER pxf_filter_push_down_server_ssl
    OPTIONS (resource 'dummy_path', format 'filter', delimiter ',');
SELECT t0, a1 FROM test_filter;
 t0 | a1 
----+----
    |  0
 B  |   
 C  |  2
 D  |  3
 E  |  4
 F  |  5
 G  |  6
 H  |  7
 I  |  8
 J  |  9
(10 rows)

ALTER FOREIGN DATA WRAPPER pxf_filter_push_down_fdw_ssl
    OPTIONS(ADD pxf_ssl_verify_peer 'false');
-- Check writable table
CREATE FOREIGN DATA WRAPPER file_pxf_fdw_ssl
    HANDLER pxf_fdw_handler
    VALIDATOR pxf_fdw_validator
    OPTIONS(protocol 'file', mpp_execute 'all segments',
        pxf_protocol 'https',
        pxf_ssl_cacert '/opt/ssl/certs/ca-cert',
        pxf_ssl_cert   '/opt/ssl/certs/pxf-client.pem',
        pxf_ssl_cert_type 'PEM',
        pxf_ssl_key '/opt/ssl/certs/pxf-client.key'
    );
CREATE SERVER pxf_file_server_ssl
    FOREIGN DATA WRAPPER file_pxf_fdw_ssl OPTIONS(
    config 'default');
CREATE USER MAPPING FOR CURRENT_USER SERVER pxf_file_server_ssl;
CREATE FOREIGN TABLE test_file2 (
    id int,
    descr text
) SERVER pxf_file_server_ssl
OPTIONS (resource 'fdw_file', format 'csv', delimiter ',');
INSERT INTO test_file2 select id, md5('hello'||id::text)as descr from generate_series(1,10) as id;
WARNING:  skipping "test_file2" --- cannot analyze this foreign table
\! find /tmp/pxf/fdw_file -type f -name \[0-9]\* | xargs cat | LC_ALL=C sort -t',' -k1,1n
1,203ad5ffa1d7c650ad681fdff3965cd2
2,6e809cbda0732ac4845916a59016f954
3,7ce8be0fa3932e840f6a19c2b83e11ae
4,a75f2192bae11cb76cdcdada9332bab6
5,ebde9cc9540087b9688fdb470fa20f17
6,5726012822477f24fe999a1f7223c82a
7,2b4c43e3b1102b2e8492eebd97c06c58
8,f70109ba33bf1860001f89998581df2a
9,1c52d3d0503eca3c4a6db73af6d428f1
10,75628fce248a7e55e27335a60900ebab
